torch/distributed/device_mesh.py

```diff
@@ -1233,6 +1236,97 @@ def _concatenate(device_mesh_list: list["DeviceMesh"]) -> "DeviceMesh":
             res_mesh._dim_group_names = concat_dim_group_name
             return res_mesh
 
+        def __getstate__(self):
+            """
+            Returns the state of the DeviceMesh as a dictionary for serialization,
+            which contains all necessary information to reconstruct the DeviceMesh.
+            """
+            state: dict[str, Any] = {
+                "device_type": self._device_type,
+                "rank_map": self._rank_map,
+                "layout": self._layout,
+                "mesh_dim_names": self._mesh_dim_names,
+                "thread_id": self._thread_id,
+                "coordinate_on_dim": getattr(self, "_coordinate_on_dim", None),
+            }
+
+            # Serialize root_mesh if it exists
+            # To avoid infinite recursion (root -> child -> root), only serialize if this is not the root
+            if self._root_mesh is not None:
+                state["root_mesh"] = self._root_mesh.__getstate__()
+            else:
+                state["root_mesh"] = None
+
+            # Serialize flatten_mapping
+            flatten_mapping: dict[str, Any] = {}
+            for mesh_name, mesh in self._flatten_mapping.items():
+                flatten_mapping[mesh_name] = mesh.__getstate__()
+            state["flatten_mapping"] = flatten_mapping
+
+            if not self.decouple_backend_at_save and hasattr(self, "_dim_group_names"):
+                logger.warning(
+                    "Save device mesh via torch.save with pg names and will be deprecated in PT 2.11. "
+                    "Users are welcome to use Distributed checkpoint (DCP) or re-create pgs in the same order"
```

Comment:
This comment is probably not detailed enough to be helpful.
* welcome to use DCP: to be clear, this suggestion is that the user rewrite their flow, though, we should at least point to a doc or tutorial 
* re-create pgs in the same order: this suggestion is not detailed enough to be actionable IMO. 

How about this?
"Starting in PyTorch 2.11, torch.save will save a DeviceMesh without including ProcessGroup information, and loading a saved DeviceMesh will require manually recreating the same configuration of ProcessGroups and binding them to the loaded DeviceMesh.  See `<link to example>` for more information on how to do this.  Alternatively, use DCP `<link to tutorial>` to save and load DTensors in a format that supports resharding and can be loaded on a different mesh configuration."